1-foundations/project-1-schelling-model/Project 1 - Schelling's model of segregation.ipynb

```diff
@@ -0,0 +1,676 @@
+    "\n",
+    "In the example below, neighborhoods are composed by the 1 x 1 sub-grid surrounding a particular location. Note that empty slots are not considered when calculating the denominator of the similarity score."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "![similarity](img/similarity-schelling.png)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "When an agent is dissatisfied, they will seek to move to another location that could be a better fit for the minimum similarity they want. The location they will move to is one among all the locations available at the moment that meet the minimum similarity score (_t_). Specifically, they will select the one that is closest to them. If more than acceptable location is closest, then they will go for the one with the highest similarity score."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "![moving](img/moving-schelling.png)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The simulation of the model is divided in rounds. When a round starts, the model iterates once through every agent evaluating their satisfaction and moving them to another spot if it applies. A round finishes when all of the agents have been looped through once, executing all the moves that applied. A new round starts when this process is repeated. The simulation finishes when the city achieves convergence (no one moved in an entire round) or a maximum number of rounds have been finalized.\n",
+    "\n",
+    "You might note that as a result of the moves in a round some agents that were initially dissatisfied will become satisfied as their neighborhoods change, or vice versa. We will not take care of these cases immediately, but in the next round of the model simulation. The instructions for that part of the implementation will be explained below."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Part 2: Python implementation\n",
+    "\n",
+    "This is how our specific implementation of Schelling's model will work:\n",
+    "\n",
+    "1. The initial state of the city (the grid) will be read from pre-defined files\n",
+    "1. When a round of the simulation begins, start the iteration with the square at the top left and perform the following operations:\n",
+    "    - Evaluate if the agent is dissatisfied\n",
+    "    - If they are:\n",
+    "        - Move the agent to its corresponding lot applying the moving rule\n",
+    "        - Update the city\n",
```

Comment:
```suggestion
    "1. When a round of the simulation begins, start the iteration with the square at the top left and perform the following operations:\n",
    "    - Evaluate if there is an agent in the square\n",
    "    - If there is one:\n",
    "        - Evaluate if the agent is dissatisfied\n",
    "        - If they are:\n",
    "            - Move the agent to its corresponding lot applying the moving rule\n",
    "            - Update the city\n",
```

Reply:
I pushed other changes before committing the suggestions, but recreated them and pushed in b9d6c24